src/Framework/NativeMethods.cs

```diff
@@ -1035,6 +1042,60 @@ internal static MemoryStatus GetMemoryStatus()
         return null;
     }
 
+    internal static bool ExistAndHasContent(string path)
+    {
+        var fileInfo = new FileInfo(path);
+
+        return fileInfo.Exists && (fileInfo.Length > 0 || IsSymLink(fileInfo));
```

Comment:
If it's a symlink, shouldn't we still check that the thing it points to is nonempty?

Reply:
Good point.
Added support for traversal to final link target and checking it's existence and nonemptiness.